src/coreclr/jit/fgwasm.cpp

```diff
@@ -0,0 +1,813 @@
+
+            // We ignore back edges; they don't inspire blocks.
+            //
+            if (succNum <= cursor)
+            {
+                JITDUMP("Backedge " FMT_BB "[%u] -> " FMT_BB "[%u]\n", block->bbNum, cursor, succ->bbNum, succNum);
+
+                // The backedge target should be a loop header.
+                // (TODO: scan loop stack to verify the loop is on the stack?)
+                //
+                // Note we currently bail out way above if there are any irreducible loops.
+                //
+                assert(m_loops->GetLoopByHeader(succ) != nullptr);
+                continue;
+            }
+
+            // Branch to next needs no block, unless this is a switch
+            // (eventually when we leave the default on the switch we can remove this).
+            //
+            if ((succNum == (cursor + 1)) && !block->KindIs(BBJ_SWITCH))
+            {
+                continue;
+            }
+
+            // Branch to cold block needs no block (presumably something EH related).
+            // Eventually we need to case these out and handle them better.
+            //
+            if (succNum >= numHotBlocks)
+            {
+                continue;
+            }
+
+            // See if we already have a block that ends at this point and starts before.
+            //
+            WasmInterval* const existingBlock = scratch[succNum];
+
+            if (existingBlock != nullptr)
+            {
+                // If so we don't need to track this branch.
+                //
+                JITDUMP("Subsumed " FMT_BB "[%u] -> " FMT_BB "[%u]\n", block->bbNum, cursor, succ->bbNum, succNum);
+                assert(existingBlock->Start() <= cursor);
+                continue;
+            }
+
+            // Non-contiguous, non-subsumed forward branch
+            //
+            WasmInterval* const branch = WasmInterval::NewBlock(this, block, initialLayout[succNum]);
+            intervals.push_back(branch);
+
+            // Remember an interval end here
+            //
+            scratch[succNum] = branch;
+        }
+    }
+
+    // Display the raw intervals...
+    //
+    JITDUMP("\n-------------- Initial set of wasm intervals\n");
+    for (WasmInterval* iv : intervals)
+    {
+        JITDUMPEXEC(iv->Dump());
+    }
+    JITDUMP("--------------\n\n");
+
+    // -----------------------------------------------
+    // (3) Find intervals that overlap
+    //
+    // See if this interval conflicts with any other. If so,
+    // add the interval to that intervals conflict set, and return
+    // the conflict set for further resolution.
+    //
+    // Since this is only looking at prior intervals it could be
+    // merged with (2) above.
+    //
+    auto resolve = [&intervals](WasmInterval* const iv) {
+        for (WasmInterval* ip : intervals)
+        {
+            // We only need to consider intervals that start at the same point or earlier.
+            //
+            if (ip == iv)
```

Comment:
Nit: it is surprisingly difficult on the eye to distinguish between `ip`/`iv`/`ic`. They may benefit from longer names.